Feat: multiple templates rewrites plp (#1157)
vtexdocs/dev-portal-content: docs/faststore/docs/headless-cms/multiple-page-template.mdx

- Department template: `/office` alternative for products within a department, ignoring category.
- Generic PLP template: Used if no other matches are found.

#### Using Rewrites with PLP template

> FastStore enables the use of [rewrites from Next.js](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites) to create a custom route.
> It's possible to enable the multiple page template feature for the custom route as well by adding your rewrites function into the `faststore.config.js` file as the Next.js doc suggests:

Reviewer: ```suggestion
FastStore allows you to create custom routes using [Next.js rewrites](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites).
It's possible to enable the Multiple page template feature for the custom route as well by adding your rewrites function into the `faststore.config.js` file as suggested in the [Next.js documentation](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites):
```

> 

Reviewer: ```suggestion
1. Add your rewrite function to faststore.config.js:

```

```js
async rewrites() {
  return [
    {
      source: '/my-office',
      destination: `/office`,
    },
  ]
},
```

> 

Reviewer: ```suggestion

This rewrites the URL `/my-office` to the existing route `/office`, effectively using the `/office` template without modifying the displayed URL. This allows you to map the template in the Headless CMS to the `/office` route.

```

> or

Reviewer: ```suggestion
2. Alternatively, use the extended format:
```

```js
async rewrites() {
  return {
    beforeFiles: [
      {
        source: '/my-office',
        destination: `/office`,
      },
    ],
    afterFiles: [],
    fallback: [],
  }
},
```

In the example above, the new `source` route `/my-office` will be rewritten to the existing `destination` route `/office`, using the `/office` template without changing the URL.

> In that way, you can use the existing `destination` route `/office` to map this template in the hCMS.

Reviewer: ```suggestion
In that way, you can use the existing `destination` route `/office` to map this template in the [Headless CMS](https://developers.vtex.com/docs/guides/faststore/headless-cms-overview).
```

> In the example above, the new `source` route `/my-office` will be rewritten to the existing `destination` route `/office`, using the `/office` template without changing the URL.
> In that way, you can use the existing `destination` route `/office` to map this template in the hCMS.

Reviewer: ```suggestion
This format offers more flexibility for advanced rewrite scenarios.
```

Reviewer: Maybe this sentence for the other code?

> From the version `3.0.X` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.

Author: 🎗️ update the version after FastStore generate the new version

> From the version `3.0.61` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.

Reviewer: ```suggestion
From the version `3.0.61` of the `@faststore/core` package, it's also possible to use the `source` route in the Headless CMS to map the template. To update your store's `@faststore/core` package to the `3.0.61`, run the following command:
```

> Update your store’s `@faststore/core` package to the `3.0.61` by running the following command:

Reviewer: ```suggestion
```

```bash

> yarn add @faststore/core@3.0.X

Author: 🎗️ update the version after FastStore generate the new version

> From the version `3.0.X` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.
> 
> Update your store’s `@faststore/core` package to the `3.0.X` by running the following command:
> 
> ```bash
> yarn add @faststore/core@3.0.X

Author: ```suggestion
From the version `3.0.61` of the `@faststore/core` package, it's also possible to use the `source` route in the hCMS to map the template.

Update your store’s `@faststore/core` package to the `3.0.61` by running the following command:

```bash
yarn add @faststore/core@3.0.61
```

> ```

Reviewer: Após esse comando, devo rodar `yarn dev` ?

Author: yes, but this is implicit if you are using local development.

- Generic PLP template: Used if no other matches are found.

#### Using Rewrites with PLP template

FastStore allows you to create custom routes using [Next.js rewrites](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites).
It's possible to enable the Multiple page template feature for the custom route as well by adding your rewrites function into the `faststore.config.js` file as suggested in the [Next.js documentation](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites):
1. Add your rewrite function to faststore.config.js:

```js
async rewrites() {
  return [
    {
      source: '/my-office',
      destination: `/office`,
    },
  ]
},
```

This rewrites the URL `/my-office` to the existing route `/office`, effectively using the `/office` template without modifying the displayed URL. This allows you to map the template in the Headless CMS to the `/office` route.

2. Alternatively, use the extended format:

```js
async rewrites() {
  return {
    beforeFiles: [
      {
        source: '/my-office',
        destination: `/office`,
      },
    ],
    afterFiles: [],
    fallback: [],
  }
},
```

> In the example above, the new `source` route `/my-office` will be rewritten to the existing `destination` route `/office`, using the `/office` template without changing the URL.
> In that way, you can use the existing `destination` route `/office` to map this template in the [Headless CMS](https://developers.vtex.com/docs/guides/faststore/headless-cms-overview).

Reviewer: ```suggestion
This format offers more flexibility for advanced rewrite scenarios.
```
